### PR TITLE
Refactored generative models

### PR DESCRIPTION
**Pull request by laurikskl**
Sorted out exports and imports.

Created a new file to split up the structs.

Refactored in general.

### CODE REVIEW

File: src/generative_models/GenerativeModels.jl
Comment (VincentPikand) [inline src/generative_models/GenerativeModels.jl:6]: flux twice

File: src/generative_models/GenerativeModels.jl
Comment (VincentPikand) [inline src/generative_models/GenerativeModels.jl:7]: not required

File: src/generative_models/encoders.jl
Comment (VincentPikand) [inline src/generative_models/encoders.jl:1]: empty line here?

File: src/generative_models/encoders.jl
Comment (VincentPikand) [inline src/generative_models/encoders.jl:33]: pretty sure Chain and Dense are from `Flux`

File: src/generative_models/vae_params.jl
Comment (VincentPikand) [inline src/generative_models/vae_params.jl:19]: I feel like this could be a part of `vae.jl`, but I'll leave it up to you.

**Review by VincentPikand — CHANGES_REQUESTED** [review]
Please go over all namespace prefixes again, I marked out a few I found.

File: src/generative_models/GenerativeModels.jl
Comment (laurikskl) [reply src/generative_models/GenerativeModels.jl:7]: Flux.Losses you think is not required?

File: src/generative_models/encoders.jl
Comment (laurikskl) [reply src/generative_models/encoders.jl:1]: Interesting that the formatter didnt complain, thanks

File: src/generative_models/GenerativeModels.jl
Comment (RaunoArike) [reply src/generative_models/GenerativeModels.jl:7]: Since you write `Flux.Losses.mse` fully out in `vae_params.jl`, I also think just importing Flux would be enough

File: src/generative_models/encoders.jl
Comment (RaunoArike) [inline src/generative_models/encoders.jl:18]: Noting just in case that these should also be changed to Flux.Dense

File: src/generative_models/vae_params.jl
Comment (RaunoArike) [reply src/generative_models/vae_params.jl:19]: I also think it's fine to have this in `vae.jl`, but check with Patrick and decide based on that

**Review by RaunoArike — CHANGES_REQUESTED** [review]
Looks fine overall, but I'll wait with approving until you've implemented the changes requested by Vincent

File: src/generative_models/encoders.jl
Comment (laurikskl) [reply src/generative_models/encoders.jl:18]: Yep they've been changed, I was a bit cofnused with the flux/flux.losses thing

**Review by RaunoArike — APPROVED** [review]

**Comment by VincentPikand** [comment]
Will approve when you've gotten an answer from Patrick.

**Review by VincentPikand — APPROVED** [review]

**Pull request merged**
